core/src/main/java/org/apache/calcite/avatica/remote/looker/LookerRemoteMeta.java

```diff
@@ -0,0 +1,438 @@
+    HostnameVerifier trustAllHostNames = (hostname, session) -> true;
+    try {
+      SSLContext sc = SSLContext.getInstance("SSL");
+      sc.init(null, trustAllCerts, new java.security.SecureRandom());
+      connection.setSSLSocketFactory(sc.getSocketFactory());
+      connection.setHostnameVerifier(trustAllHostNames);
+    } catch (NoSuchAlgorithmException | KeyManagementException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * A regrettable necessity. The Kotlin SDK relies on an outdated Ktor HTTP client based on Kotlin
+   * Coroutines which are difficult to work with in Java. Here we make a HTTP client to handle the
+   * request and input stream ourselves. This adds complexity that would normally be handled by the
+   * SDK. We should revisit this once the SDK has built-in streams.
+   *
+   * TODO https://github.com/looker-open-source/sdk-codegen/issues/1341:
+   *  Add streaming support to the Kotlin SDK.
+   */
+  protected InputStream makeRunQueryRequest(String url) throws IOException {
+    AuthSession authSession = getSdk().getAuthSession();
+    Transport sdkTransport = authSession.getTransport();
+
+    // makes a proper URL from the API endpoint path as the SDK would.
+    String endpoint = sdkTransport.makeUrl(url, Collections.emptyMap(), null);
+    URL httpsUrl = new URL(endpoint);
+    HttpsURLConnection connection = (HttpsURLConnection) httpsUrl.openConnection();
+
+    // WARNING: You should only set `verifySSL=false` for local/dev instances!!
+    if (!sdkTransport.getOptions().getVerifySSL()) {
+      trustAllHosts(connection);
+    }
+
+    // timeout is given as seconds
+    int timeout = sdkTransport.getOptions().getTimeout() * 1000;
+    connection.setReadTimeout(timeout);
+    connection.setRequestMethod("GET");
+    connection.setRequestProperty("Accept", "application/json");
+    connection.setDoOutput(true);
+
+    // Set the auth header as the SDK would
+    connection.setRequestProperty("Authorization",
+        "token " + authSession.getAuthToken().getAccessToken());
+
+    int responseCode = connection.getResponseCode();
+    if (responseCode == 200) {
+      // return the input stream to parse from.
+      return connection.getInputStream();
+    } else {
+      throw new IOException("HTTP request failed with status code: " + responseCode);
+    }
+  }
+
+  private void seekToRows(JsonParser parser) throws IOException {
+    while (parser.nextToken() != null && !ROWS_KEY.equals(parser.currentName())) {
+      // move position to start of `rows`
+    }
+  }
+
+  private void seekToValue(JsonParser parser) throws IOException {
+    while (parser.nextToken() != null && !VALUE_KEY.equals(parser.currentName())) {
+      // seeking to `value` key for the field e.g. `"rows": [{"field_1": {"value": 123 }}]
+    }
+    // now move to the actual value
+    parser.nextToken();
+  }
+
+  private void putExceptionOrFail(BlockingQueue queue, Exception e) {
```

Comment:
I guess `BlockingQueue` started out polymorphic, containing either `Frame` or `Exception` instances, but we shouldn't use raw generic collections, it should be updated to `BlockingQueue<FrameEnvelope>` throughout. It would eliminate some explicit casts too.